### PR TITLE
Stats: Add Tracks to MobilePromoCard component

### PR DESCRIPTION
**Pull request by a8ck3n**
#### Proposed Changes

Adds support for Tracks to the various links. We do not track the QR code as there isn't any browser event for someone scanning a QR code. The tracked events are:

**FOR JETPACK:**
* `calypso_stats_mobile_cta_jetpack_view` — The Jetpack CTA is rendered.
* `calypso_stats_mobile_cta_jetpack_click` — User clicks link to jetpack.com/apps
* `calypso_stats_mobile_cta_jetpack_apple_click` — User clicks App Store badge for Jetpack app
* `calypso_stats_mobile_cta_jetpack_google_click` — User clicks Google Play badge for Jetpack app

**FOR WOO**
* `calypso_stats_mobile_cta_woo_view` — The Woo CTA is rendered.
* `calypso_stats_mobile_cta_woo_click` — User clicks link to woo.com/mobile
* `calypso_stats_mobile_cta_woo_apple_click` — User clicks App Store badge for Woo app
* `calypso_stats_mobile_cta_woo_google_click` — User clicks Google Play badge for Woo app

Additionally, I've removed the raw URLs in favour of the jetpack.com redirects. This allows us to update the URLs server-side without having to make code changes. Applies to QR codes as well.

#### Testing Instructions

* Check out this branch.
* Run `yarn workspace @automattic/components run storybook`.
* Visit the Mobile Promo Card entry.
* Confirm the icons, text, and QR codes display correctly.
* Confirm the QR codes link out correctly.
* Use the Chrome dev tools and change to the "iPhone 12 Pro" setting. Refresh the view and confirm you get an App Store badge that links to the app page.
* Change to the "Pixel 5" setting and refresh again. Confirm you get a Play Store badge and that it's wired up correctly.
* Wait 24 hours to confirm our events are being recorded correctly?!

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #70463.

### CODE REVIEW

**Comment by jeffgolenski** [comment]
Thanks @a8ck3n those tracks events work fine for me! Are we also able to build in specific "impression" events so we can be very targeted about measuring views -> clicks. I don't want to rely on `calypso_page_view` or whatever events... I don't think they're specific enough when it comes to measuring CR. 

Just as a note for the whole team @keoshi @jsnmoon @dognose24 @grzegorz-cp - I'm continuing to work on the data mapping stuff and we will definitely be having a larger convo about all this during the maintenance project! Thanks y'allll

File: packages/components/src/mobile-promo-card/index.tsx
Comment (grzegorz-cp) [inline packages/components/src/mobile-promo-card/index.tsx:188]: Previous `Woo` and `Jetpack` strings were also links. Is this change intentional?

**Review by grzegorz-cp — APPROVED** [review]
The code looks good. I checked everything apart from the last step required waiting for 24h to see the events. Great description of testing steps.

File: packages/components/src/mobile-promo-card/index.tsx
Comment (a8ck3n) [reply packages/components/src/mobile-promo-card/index.tsx:188]: Yes, this was intentional in response to some design feedback.

**Comment by a8ck3n** [comment]
> Thanks @a8ck3n those tracks events work fine for me! Are we also able to build in specific "impression" events so we can be very targeted about measuring views -> clicks. I don't want to rely on `calypso_page_view` or whatever events... I don't think they're specific enough when it comes to measuring CR.

Hey @jeffgolenski. I'm not sure I follow. Are you looking for a Tracks event specific to the Traffic page in addition to the click events? Does that allow you to get the measurement you need?

**Comment by jeffgolenski** [comment]
> Hey @jeffgolenski. I'm not sure I follow. Are you looking for a Tracks event specific to the Traffic page in addition to the click events? Does that allow you to get the measurement you need?

Sorry if I'm not being clear @a8ck3n! In addition to all the click events you've built in, I'd also like us to build in specific events for the banners when they're "seen" or loaded. As you mention, it would be best if we can easily attribute those impressions to specific pages

Couple of examples for why we should have this:
1) I want to see how many times the banner was seen on the Traffic page, and out of those views, how many people clicked it.
2) I want to compare the conversion rates of these banner clicks / taps on the traffic page and insights page.

So I suppose what would be really nice to see are some events like:
`calypso_stats_mobile_cta_jetpack_view` along with `calypso_stats_mobile_cta_jetpack_click` and then when I build a funnel with those two events I can easily segment by page or device. 

The reason for this is that in all my years in tracks, if we don't have highly specific events for views, then we have to finagle with using `calypso_page_view` and insert specific paths into the funnels and it gets really messy and you need to know exactly what you're looking for.

**Comment by a8ck3n** [comment]
> Sorry if I'm not being clear @a8ck3n! In addition to all the click events you've built in, I'd also like us to build in specific events for the banners when they're "seen" or loaded. As you mention, it would be best if we can easily attribute those impressions to specific pages

Sounds good to me @jeffgolenski! I'll get started on those extra Tracks events today.

File: packages/components/src/mobile-promo-card/index.tsx
Comment (jsnmoon) [inline packages/components/src/mobile-promo-card/index.tsx:57]:
```suggestion
	const index = slugs.indexOf( slug );
	if ( ! slugs.includes( slug ) ) {
```

File: packages/components/src/mobile-promo-card/index.tsx
Comment (jsnmoon) [inline packages/components/src/mobile-promo-card/index.tsx:54]: We could define this as a constant outside of the function, like so:

```js
	const REDIRECTABLE_SLUGS = [
		URL_JETPACK_A8C,
		URL_JETPACK_APPLE,
		URL_JETPACK_GOOGLE,
		URL_JETPACK_QRCODE,
		URL_WOO_A8C,
		URL_WOO_APPLE,
		URL_WOO_GOOGLE,
		URL_WOO_QRCODE,
	];
```

File: packages/components/src/mobile-promo-card/index.tsx
Comment (jsnmoon) [inline packages/components/src/mobile-promo-card/index.tsx:43]: Why is the function name suffixed with `Private`? Since the function's not being exported, its scope is already limited to this file.

File: packages/components/src/mobile-promo-card/index.tsx
Comment (jsnmoon) [inline packages/components/src/mobile-promo-card/index.tsx:177]: We'll want these events to fire only upon mounting using [useEffect](https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect), like so:

```suggestion
	// Track "views" of this card.
	useEffect( () => {
		if ( isWoo ) {
			recordTracksEventPrivate( EVENT_WOO_VIEW );
		} else {
			recordTracksEventPrivate( EVENT_JETPACK_VIEW );
		}
	}, [ isWoo ] );
```

File: packages/components/src/mobile-promo-card/index.tsx
Comment (jsnmoon) [inline packages/components/src/mobile-promo-card/index.tsx:75]: This array could also be defined as a constant outside the function.

File: packages/components/src/mobile-promo-card/index.tsx
Comment (jsnmoon) [inline packages/components/src/mobile-promo-card/index.tsx:28]: If we defined these as a [standard Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), we wouldn't need to create `slugs` or `events` arrays below. We could use [`Map.prototype.has`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) to check membership/validity.

**Review by jsnmoon — COMMENTED** [review]
This PR needs a trunk merge! The overall approach looks solid, so let us know when it's ready for review.

File: packages/components/src/mobile-promo-card/index.tsx
Comment (a8ck3n) [reply packages/components/src/mobile-promo-card/index.tsx:57]: Thanks for the heads up. This is definitely the clearer option. I have a smaller follow-up PR and I'll include this change there.

File: packages/components/src/mobile-promo-card/index.tsx
Comment (a8ck3n) [reply packages/components/src/mobile-promo-card/index.tsx:43]: The goal with this is to help with readability in search results. I think the better option is probably to rename it to be slightly more specific though. So maybe something like `getRedirectUrlFromSlug()`. I'll give this another look.

File: packages/components/src/mobile-promo-card/index.tsx
Comment (a8ck3n) [reply packages/components/src/mobile-promo-card/index.tsx:28]: I spent some time tinkering with a Map (in response to this feedback) but I couldn't quite get it working the way I wanted. While it eliminated the need for the array of CONST values, it means I then had to use raw strings at the call sites. I tried something like this:

```js
// set up a Map of known slugs
const kvArray = [
  [ 'URL_JETPACK_A8C', 'calypso-stats-mobile-cta-jetpack-link' ],
  ['URL_WOO_A8C', 'calypso-stats-mobile-cta-woo-link' ]
];
const slugMap = new Map( kvArray );
// check for valid slug
function getURLForSlug( slug ) {
  if ( slugMap.has( slug ) ) {
    console.log( 'valid slug' );
  }
  return '';
}
// at call site. need to use raw strings?
const url = getURLForSlug( 'URL_JETPACK_A8C' );
console.log( url );
```

I ended up going with a configuration Object instead. It let me test for validity and avoid raw slugs at the call sites. I'll ping you from the updated PR so you can have a look. Also, please let me know if I overlooked something in my Map approach. Thanks @jsnmoon!

File: packages/components/src/mobile-promo-card/index.tsx
Comment (a8ck3n) [reply packages/components/src/mobile-promo-card/index.tsx:57]: Addressed in #70926.

File: packages/components/src/mobile-promo-card/index.tsx
Comment (a8ck3n) [reply packages/components/src/mobile-promo-card/index.tsx:54]: Addressed in #70926.

File: packages/components/src/mobile-promo-card/index.tsx
Comment (a8ck3n) [reply packages/components/src/mobile-promo-card/index.tsx:43]: Addressed in #70926.

File: packages/components/src/mobile-promo-card/index.tsx
Comment (a8ck3n) [reply packages/components/src/mobile-promo-card/index.tsx:28]: Addressed in #70926.

Comment (jsnmoon) [reply packages/components/src/mobile-promo-card/index.tsx:28]: Happy to help! And you pretty much nailed it; this was what I was thinking (using your code snippet above as the basis):

```js
const slugMap = new Map( [
	[ 'jetpack-a8c', 'calypso-stats-mobile-cta-jetpack-link' ],
	[ 'woo-a8c', 'calypso-stats-mobile-cta-woo-link' ],
] );

function getURLForSlug( slug ) {
	if ( slugMap.has( slug ) ) {
		console.log( 'valid slug' );
	}
	return '';
}

const url = getURLForSlug( 'jetpack-a8c' );
console.log( url );
```

Note that `map.get( 'jetpackA8C' )` is not much different from `object.jetpackA8C`, which is the same as `object[ 'jetpackA8C' ]`.

**Comment by jeffgolenski** [comment]
@a8ck3n I noticed you're still waiting on my review for this, but you definitely don't want a code review from me unless it's CSS. 😆

**Comment by a8ck3n** [comment]
> @a8ck3n I noticed you're still waiting on my review for this, but you definitely don't want a code review from me unless it's CSS. 😆

Sorry @jeffgolenski, I tried removing the Needs Review label but matticbot keeps adding it back. I'm going to close this as I'm breaking the PR into smaller chunks to get around the issue where the tests keep failing.

**Comment by jeffgolenski** [comment]
@a8ck3n haha you're good my friend! I just didn't want to hold you up!

**Pull request closed without merging**
